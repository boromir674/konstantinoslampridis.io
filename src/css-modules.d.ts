// Global type declarations for CSS modules
declare module '*.module.css' {
  const classes: { readonly [key: string]: string };
  export = classes;
}

declare module '*.module.scss' {
  const classes: { readonly [key: string]: string };
  export = classes;
}

declare module '*.module.sass' {
  const classes: { [key: string]: string };
  export default classes;
}
