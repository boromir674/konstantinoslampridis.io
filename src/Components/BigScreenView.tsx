import { FC } from "react";
import styled from "@emotion/styled";

import TopHeaderPane, { TopHeaderPaneProps } from "./TopHeaderPane";
import BottomFooterPane, { BottomFooterPaneProps } from "./BottomFooterPane";
import VerticalMainPane, { AppVerticalMainPaneProps } from "./VerticalMainPane";
import VerticalSidePane, { AppVerticalSidePaneProps } from "./VerticalSidePane";

interface BigScreenViewContainerProps {
  children?: React.ReactNode;
  theme: {
    containerBackgroundColor: string;
  };
}

interface BigScreenViewProps {
  theme: {
    containerBackgroundColor: string;
    topHeaderPane: TopHeaderPaneProps["theme"];
    verticalSidePane: AppVerticalSidePaneProps["theme"];
    verticalMainPane: AppVerticalMainPaneProps["theme"];
    bottomFooterPane: BottomFooterPaneProps["theme"];
  };
  data: {
    topHeaderPane: TopHeaderPaneProps["data"];
    verticalSidePane: AppVerticalSidePaneProps["data"];
    verticalMainPane: AppVerticalMainPaneProps["data"];
    // bottomFooterPane: BottomFooterPaneProps['data'];
  };
}

const BigScreenViewContainer = styled.div<BigScreenViewContainerProps>`
  display: grid;
  grid-template-columns: 1fr 3fr;
  //   grid-template-rows: 1fr 3fr 1fr;
  grid-template-rows: 1fr 3fr 1fr;
  grid-template-areas:
    "Header Header"
    "Side Main"
    "Footer Footer";
  height: 100vh;
  width: 100vw;
  // background: #ffecb3;
  background: ${(props) => props.theme.containerBackgroundColor};
//   background: "inherit";
`;

const BigScreenView: FC<BigScreenViewProps> = ({
  theme: {
    containerBackgroundColor,
    topHeaderPane,
    verticalSidePane,
    verticalMainPane,
    bottomFooterPane,
  },
  data: {
    topHeaderPane: topHeaderPaneData,
    verticalSidePane: verticalSidePaneData,
    verticalMainPane: verticalMainPaneData,
    // bottomFooterPane: bottomFooterPaneData,
  },
}) => {
  return (
    <BigScreenViewContainer theme={{containerBackgroundColor: containerBackgroundColor}}>
      <TopHeaderPane theme={topHeaderPane} data={topHeaderPaneData} />
      <VerticalSidePane theme={verticalSidePane} data={verticalSidePaneData} />
      <VerticalMainPane theme={verticalMainPane} data={verticalMainPaneData} />
      <BottomFooterPane theme={bottomFooterPane} />
    </BigScreenViewContainer>
  );
};

export default BigScreenView;
