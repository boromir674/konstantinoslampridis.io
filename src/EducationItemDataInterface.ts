/** 
* A Person's Education Item information.
* @summary This is how the App represents a Person's Education item.
* @interface EducationItemData
* @property {string} degree_title - The title of the degree.
* @property {string} university_name - The name of the university.
* @property {string} location - The location of the university.
* @property {string} duration - The duration of the degree/studies (ie '2018', '2014 - 2019').
* @property {string} thesis_title - The title of the thesis.
* @property {string[]} topics - List of courses/topics in the degree curriculum.
*/
interface EducationItemData {
    degree_title: string;
    university_name: string;
    location: string;
    duration: string;
    thesis_title: string;
    topics: string[];
  }

export default EducationItemData;
