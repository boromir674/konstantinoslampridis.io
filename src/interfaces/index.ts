import type { LayoutInterface, LayoutsObject } from './GridLayoutInterfaces';
import IconStyles from './SVGIconStyles';

export type { IconStyles, LayoutInterface, LayoutsObject };
