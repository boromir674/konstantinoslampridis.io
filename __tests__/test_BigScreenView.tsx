/**
 * @jest-environment jsdom
 */

// Test Helpers Imports
import React from "react";
import { Context as ResponsiveContext } from "react-responsive";
import { render } from "@testing-library/react";

// Business-logic Imports
import BigScreenView, { BigScreenViewProps } from "../src/Components/BigScreenView";
import { lightTheme } from '../src/theme';

// TEST DATA
const DATA = {
  topHeaderPane: {
    sections: [
      { name: "Home", to_element_id: "home-section" },
      { name: "Portfolio", to_element_id: "portfolio-section" },
      { name: "Professional", to_element_id: "professional-section" },
    ],
    onToggle: (active: boolean) => {
      console.log("active: ", active);
    },
    // starting position of toggle, if true it is on the right
    active: true,
    // false -> left, true -> right
  },
  verticalSidePane: {
    personal: {
      name: "John Doe",
      email: "jd@email.io",
      github: "github.com/john-doe",
      gitlab: "gitlab.com/john-doe",
      linkedin: "linkedin.com/in/john-doe",
    },
    education: [
      {
        degree_title: "MSc in Artificial Intelligence",
        university_name: "University of Amsterdam",
        location: "Amsterdam, Netherlands",
        duration: "2014 - 2019",
        thesis_title: "Political Spectrum Aware Topic Model",
        topics: ["ML", "CV", "RL", "NLP"],
      },
      {
        degree_title: "BSc in Applied Informatics",
        university_name: "University of Macedonia",
        location: "Thessaloniki, Greece",
        duration: "2008 - 2013",
        thesis_title: "Computational Analysis of Simplex Points Algorithm",
        topics: [
          "Operating Systems",
          "Linear Programming",
          "Discreet Mathematics",
        ],
      },
    ],
  },
  verticalMainPane: {
    data: {
      introduction: {
        name: "John Doe",
      },
      professional: [
        {
          title: "Software Engineer",
          company: "GG Navi",
          location: "Mountain View, CA",
          duration: "Sep 2022 - May 2023",
          description: "I worked at GG Navi as a Software Engineer.",
          activities: [
            "I worked at GG Navi as a Software Engineer.",
            "other activity",
          ],
          technology_tags: ["python", "docker"],
        },
        {
          title: "Software Engineer",
          company: "GG Navi",
          location: "Mountain View, CA",
          duration: "Sep 2022 - May 2023",
          description: "I worked at GG Navi as a Software Engineer.",
          activities: [
            "I worked at GG Navi as a Software Engineer.",
            "other activity",
          ],
          technology_tags: ["python", "docker"],
        },
      ],
      portfolio: [
        // PROJECT 1
        {
          title: "Python Package Generator",
          development_period: "2020-2021",
          status: "Mature",
          description:
            "A CLI tool to generate a Python package with a single command.",
          source_code_repo: "boromir674/cookiecutter-python-package",
          resource_links: [
            {
              url: 'https://github.com/example/repo',
              type: 'github',
            },
            {
              url: 'https://example.com/docs',
              type: 'docs',
            },
            {
              url: 'https://example.com/ci-cd',
              type: 'ci/cd',
            },
          ],
          release: [
            {
              type: "pypi",
              name: "cookiecutter-python-package",
              artifact_version: "2.4.0",
              urlText: "https://pypi.org/projects/cookiecutter-python-package",
            },
            {
              // docker pull boromir674/generate-python:v2.4.0
              type: "docker",
              name: "generate-python",
              artifact_version: "v2.4.0",
              urlText: "https://pypi.org/projects/cookiecutter-python-package",
            },
            {
              type: "github",
              name: "cookiecutter-python-package",
              urlText: "https://pypi.org/projects/cookiecutter-python-package",
              artifact_version: "v2.4.0",
            },
          ],
          tags: ["Python", "CLI", "Automation", "Docker"],
        },
        // PROJECT 2
        {
          title: "Neural Style Transfer",
          development_period: "2020-2021",
          status: "Stable",
          description:
            "Re-paint any image with the style of another image (ie Van Gogh painting) using a Neural Style Transfer algorithm, accessible through an easy-to-use CLI.",
          source_code_repo: "boromir674/neural-style-transfer",
          resource_links: [
            {
              url: 'https://github.com/neural-style-transfer',
              type: 'github',
            },
            {
              url: 'https://neural-style-transfer.com/ci-cd',
              type: 'ci/cd',
            },
          ],
          release: [
            {
              type: "pypi",
              artifact_version: "0.6.1",
              name: "neural-style-transfer",
              urlText: "https://pypi.org/projects/cookiecutter-python-package",
            },
            {
              type: "docker",
              name: "nst",
              artifact_version: "v0.6.1",
              urlText: "https://pypi.org/projects/cookiecutter-python-package",
            },
            {
              type: "github",
              name: "neural-style-transfer",
              urlText: "https://pypi.org/projects/cookiecutter-python-package",
              artifact_version: "v0.6.1",
            },

          ],
          tags: [
            "Neural Style Transfer",
            "Deep Learning",
            "Python",
            "Docker",
            "CLI",
            "Sphinx",
            "PyPI",
          ],
        },
        // PROJECT 3
        {
          title: "Topic Modeling Toolkit",
          development_period: "2018-2019",
          status: "Mature",
          description:
            "A Python library for Topic Modeling providing a unified interface facilitating various research operations. It is designed to be easily extensible, allowing developers to implement their own algorithms and plug them in the toolkit.",
          source_code_repo: "boromir674/topic-modeling-toolkit",
          resource_links: [
            {
              url: 'https://github.com/neural-style-transfer',
              type: 'github',
            },
          ],
          release: [
            {
              type: "pypi",
              artifact_version: "0.5.2",
              name: "topic-modeling-toolkit",
              urlText: "https://pypi.org/projects/cookiecutter-python-package",
            },
            {
              type: "docker",
              name: "topic-modeling",
              artifact_version: "v0.5.2",
              urlText: "https://pypi.org/projects/cookiecutter-python-package",
            },
            {
              type: "github",
              name: "topic-modeling-toolkit",
              urlText: "https://pypi.org/projects/cookiecutter-python-package",
              artifact_version: "v0.5.2",
            },
          ],
          tags: [
            "Machine Learning",
            "Unsupervised Learning",
            "Topic Modeling",
            "Regression Testing",
            "Python",
            "PyPI",
            "automation",
            "CLI",
          ],
        },
        // Project with 2 Releases
        {
          title: "Project with 2 Releases",
          development_period: "2018-2019",
          status: "Mature",
          description:
            "A Python library for Topic Modeling providing a unified interface facilitating various research operations. It is designed to be easily extensible, allowing developers to implement their own algorithms and plug them in the toolkit.",
          source_code_repo: "boromir674/repo-with-2-releases",
          resource_links: [
            {
              url: 'https://github.com/repo-with-2-releases',
              type: 'github',
            },
            {
              url: 'https://repo-with-2-releases.com/ci-cd',
              type: 'ci/cd',
            },
          ],
          release: [
            {
              type: "pypi",
              artifact_version: "0.5.2",
              name: "repo-with-2-releases",
              urlText: "https://pypi.org/projects/repo-with-2-releases",
            },
            {
              type: "github",
              name: "repo-with-2-releases",
              urlText: "https://pypi.org/projects/repo-with-2-releases",
              artifact_version: "v0.5.2",
            },
          ],
          tags: [
            "Machine Learning",
            "Unsupervised Learning",
            "Topic Modeling",
            "Regression Testing",
            "Python",
            "PyPI",
            "automation",
            "CLI",
          ],
        },
      ],
    },
    // sectionIDs: {},
  },
};

// Helper Code to adapt Lib Theme to App Theme
type LibTheme = typeof lightTheme;
type AppColorTheme = BigScreenViewProps["theme"];

// compute maximum number of Releases contained in a single portfolio item
const maxNumberOfReleasesPerPortfolioItems = DATA.verticalMainPane.data.portfolio.reduce(
    (acc, { release = [] }) => Math.max(acc, release.length),
    0
);
// compute maximum number of links contained in a single portfolio item
const maxNumberOfLinksPerPortfolioItems = DATA.verticalMainPane.data.portfolio.reduce(
    (acc, { resource_links = [] }) => Math.max(acc, resource_links.length),
    0
);

const computeTheme = (theme: LibTheme): AppColorTheme => {
    console.log("Lib Description Theme: ", theme.portfolio.item.projectDescription);
    const adaptTheme = (theme: LibTheme) => {
        return {
            containerBackgroundColor: theme.backgroundColor,
            topHeaderPane: {
                themeSwitch: theme.themeSwitch,
                navigationBar: theme.navigationBar,
                backgroundColor: theme.topHeaderPane.backgroundColor,
            },
            verticalSidePane: {
                personalInfo: {
                    // pass Theme Personal Color Design
                    ...theme.personal,
                    // adjust interface
                    linkColor: theme.personal.urlTextColor,
                },
                education: theme.education,
            },
            verticalMainPane: {
                introduction: theme.introduction,
                professional: theme.professional,
                portfolio: {
                    ...theme.portfolio,
                    item: {
                        ...theme.portfolio.item,
                        theme: {
                            // Portfolio Project Item - Project Title and Description
                            ...theme.portfolio.item,
                            links: {
                                ...theme.portfolio.item.resourceLinks,
                                item: {
                                    ...theme.portfolio.item.resourceLinks.item,
                                    icon: {
                                        svgStyles: {
                                            // TODO: supply from theme object
                                            width: "12px",
                                            height: "12px",
                                            fill: theme.portfolio.item.resourceLinks.item.color,
                                        },
                                    },
                                },
                            },
                            // Portfolio Project Item - Software Releases
                            releases: {
                                ...theme.portfolio.item.releases,
                                headerFontFamily: theme.portfolio.item.releases.fontFamily,
                                headerColor: theme.portfolio.item.releases.color,
                                releaseButtonTheme: {
                                    ...theme.portfolio.item.releases.item,
                                    icon: {
                                        svgStyles: {
                                            // TODO: supply from theme object
                                            width: "12px",
                                            height: "12px",
                                            fill: theme.portfolio.item.releases.item.color,
                                        },
                                    },
                                },
                            },
                        },
                    },
                },
                // ...theme,
                // containerBackgroundColor: theme.backgroundColor,
            },
            bottomFooterPane: {
                ...theme.footerStyles,
                // svgStyles: {
                //   width: "15px",
                //   height: "15px",
                // }
            },
        };
    };
    const appTheme = adaptTheme(theme);

    console.log("Addapted Description Theme Artifact 1: ", appTheme.verticalMainPane.portfolio.item.projectDescription);
    console.log("Addapted Description Theme Artifact 2: ", appTheme.verticalMainPane.portfolio.item.theme.projectDescription);

    // Adapt 'icon' to 'icons' by crating an rray of the same item multiple times
    const adaptedAppTheme: AppColorTheme = {
        ...appTheme,
        verticalMainPane: {
            ...appTheme.verticalMainPane,
            portfolio: {
                ...appTheme.verticalMainPane.portfolio,
                item: {
                    ...appTheme.verticalMainPane.portfolio.item,
                    theme: {
                        ...appTheme.verticalMainPane.portfolio.item.theme,
                        links: {
                            ...appTheme.verticalMainPane.portfolio.item.theme.links,
                            item: {
                                ...appTheme.verticalMainPane.portfolio.item.theme.links.item,
                                icons: Array.from({ length: maxNumberOfLinksPerPortfolioItems }, () => appTheme.verticalMainPane.portfolio.item.theme.links.item.icon
                                ),
                            },
                        },
                        releases: {
                            ...appTheme.verticalMainPane.portfolio.item.theme.releases,
                            releaseButtonTheme: {
                                ...appTheme.verticalMainPane.portfolio.item.theme.releases.releaseButtonTheme,
                                icons: Array.from({ length: maxNumberOfReleasesPerPortfolioItems }, () => appTheme.verticalMainPane.portfolio.item.theme.releases.releaseButtonTheme.icon
                                ),
                            },
                        },
                    },
                },
            },
        },
    }
    console.log("Final Description Theme: ", adaptedAppTheme.verticalMainPane.portfolio.item.theme.projectDescription);

    return adaptedAppTheme;
};

const LIGHT_THEME: AppColorTheme = computeTheme(lightTheme);

// PROPS VALUES
const args: BigScreenViewProps = {
  // same interface as the props of the Component
  theme: LIGHT_THEME,
  data: DATA,
};

export const Light = {
  args,
};


describe("Test App width = 700", () => {

  beforeEach(() => {
    // mock ResizeObserver browser-api
    window.ResizeObserver = jest.fn().mockImplementation(() => ({
      observe: jest.fn(),
      unobserve: jest.fn(),
      disconnect: jest.fn(),
    }))
  })

  test("matches the snapshot", () => {
    const { container: mobile } = render(
      <ResponsiveContext.Provider value={{ width: 700 }}>
        <BigScreenView {...Light.args} />
      </ResponsiveContext.Provider>
    );
    expect(mobile).toMatchSnapshot();
  });
});

// describe("Test App width = 400", () => {
//   test("matches the snapshot", () => {
//     const { container: mobile } = render(
//       <ResponsiveContext.Provider value={{ width: 400 }}>
//         <IndexPage />
//       </ResponsiveContext.Provider>
//     );
//     expect(mobile).toMatchSnapshot();
//   });
// });
