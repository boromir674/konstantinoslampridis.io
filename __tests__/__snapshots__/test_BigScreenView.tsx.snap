// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App width = 700 matches the snapshot 1`] = `
<div>
  <div
    class="css-kzq97d"
  >
    <div
      class="css-1m5sbi6"
    >
      <div
        style="display: flex; flex-flow: row nowrap; cursor: pointer; width: 48px; user-select: none;"
      >
        <div
          style="padding-top: 0px; box-sizing: border-box; width: 100%; flex: 0 0 auto;"
        >
          <div
            style="width: 48px; height: 26px; border-radius: 16px; background: rgb(255, 255, 255); transition: all 100ms;"
          />
        </div>
        <div
          style="margin-left: -100%; box-sizing: border-box; width: 100%; flex: 0 0 auto;"
        >
          <div
            style="transition: all 100ms; padding-top: 4px; padding-left: 25px;"
          >
            <div
              style="width: 18px; height: 18px; size: 18px; border-radius: 16px; background-color: rgb(255, 251, 255); transition: all 100ms;"
            />
          </div>
        </div>
      </div>
      <nav
        class="css-qx08l1"
      >
        <a
          class="css-arku82"
        >
          Home
        </a>
        <a
          class="css-w0in7m"
        >
          Portfolio
        </a>
        <a
          class="css-w0in7m"
        >
          Professional
        </a>
      </nav>
    </div>
    <div
      class="css-1lxjc1s"
    >
      <div
        class="css-7xsp98"
      >
        <div
          class="css-kp9dvk"
        >
          <span
            class="css-bwb5ro"
          >
            Name:
          </span>
           
          <span
            class="css-1pgu2xd"
          >
            John Doe
          </span>
        </div>
        <div
          class="css-kp9dvk"
        >
          <span
            class="css-bwb5ro"
          >
            Email:
          </span>
           
          <span
            class="css-1pgu2xd"
          >
            jd@email.io
          </span>
        </div>
        <div
          class="css-kp9dvk"
        >
          <span
            class="css-bwb5ro"
          >
            Github:
          </span>
           
          <span
            class="css-1pgu2xd"
          >
            <a
              class="css-1gyz2k7"
              href="https://github.com/john-doe"
              rel="noopener noreferrer"
              target="_blank"
            >
              github.com/john-doe
               
              <svg
                fill="none"
                height="10px"
                stroke="#534341"
                stroke-width="5.108"
                viewBox="-2.56 -2.56 69.12 69.12"
                width="10px"
              >
                <g
                  id="SVGRepo_iconCarrier"
                >
                  <path
                    d="M55.4,32V53.58a1.81,1.81,0,0,1-1.82,1.82H10.42A1.81,1.81,0,0,1,8.6,53.58V10.42A1.81,1.81,0,0,1,10.42,8.6H32"
                  />
                  <polyline
                    points="40.32 8.6 55.4 8.6 55.4 24.18"
                  />
                  <line
                    x1="19.32"
                    x2="54.61"
                    y1="45.72"
                    y2="8.91"
                  />
                </g>
              </svg>
            </a>
          </span>
        </div>
        <div
          class="css-kp9dvk"
        >
          <span
            class="css-bwb5ro"
          >
            Gitlab:
          </span>
           
          <span
            class="css-1pgu2xd"
          >
            <a
              class="css-1gyz2k7"
              href="https://gitlab.com/john-doe"
              rel="noopener noreferrer"
              target="_blank"
            >
              gitlab.com/john-doe
               
              <svg
                fill="none"
                height="10px"
                stroke="#534341"
                stroke-width="5.108"
                viewBox="-2.56 -2.56 69.12 69.12"
                width="10px"
              >
                <g
                  id="SVGRepo_iconCarrier"
                >
                  <path
                    d="M55.4,32V53.58a1.81,1.81,0,0,1-1.82,1.82H10.42A1.81,1.81,0,0,1,8.6,53.58V10.42A1.81,1.81,0,0,1,10.42,8.6H32"
                  />
                  <polyline
                    points="40.32 8.6 55.4 8.6 55.4 24.18"
                  />
                  <line
                    x1="19.32"
                    x2="54.61"
                    y1="45.72"
                    y2="8.91"
                  />
                </g>
              </svg>
            </a>
          </span>
        </div>
        <div
          class="css-kp9dvk"
        >
          <span
            class="css-bwb5ro"
          >
            LinkedIn:
          </span>
           
          <span
            class="css-1pgu2xd"
          >
            <a
              class="css-1gyz2k7"
              href="https://linkedin.com/in/john-doe"
              rel="noopener noreferrer"
              target="_blank"
            >
              linkedin.com/in/john-doe
               
              <svg
                fill="none"
                height="10px"
                stroke="#534341"
                stroke-width="5.108"
                viewBox="-2.56 -2.56 69.12 69.12"
                width="10px"
              >
                <g
                  id="SVGRepo_iconCarrier"
                >
                  <path
                    d="M55.4,32V53.58a1.81,1.81,0,0,1-1.82,1.82H10.42A1.81,1.81,0,0,1,8.6,53.58V10.42A1.81,1.81,0,0,1,10.42,8.6H32"
                  />
                  <polyline
                    points="40.32 8.6 55.4 8.6 55.4 24.18"
                  />
                  <line
                    x1="19.32"
                    x2="54.61"
                    y1="45.72"
                    y2="8.91"
                  />
                </g>
              </svg>
            </a>
          </span>
        </div>
        <div
          class="css-kp9dvk"
        >
           
          <span
            class="css-bwb5ro"
          >
            Curricular Docs:
          </span>
           
          <div
            class="css-5l6624"
          >
            <div
              class="css-1rstcrp"
            >
              <a
                class="css-1gyz2k7"
                href="https://konstantinos-lampridis-cv-documents.s3.eu-central-1.amazonaws.com/main_resume.pdf"
                rel="noopener noreferrer"
                target="_blank"
              >
                '1 Page Resume' 
                <svg
                  fill="none"
                  height="10px"
                  stroke="#534341"
                  stroke-width="5.108"
                  viewBox="-2.56 -2.56 69.12 69.12"
                  width="10px"
                >
                  <g
                    id="SVGRepo_iconCarrier"
                  >
                    <path
                      d="M55.4,32V53.58a1.81,1.81,0,0,1-1.82,1.82H10.42A1.81,1.81,0,0,1,8.6,53.58V10.42A1.81,1.81,0,0,1,10.42,8.6H32"
                    />
                    <polyline
                      points="40.32 8.6 55.4 8.6 55.4 24.18"
                    />
                    <line
                      x1="19.32"
                      x2="54.61"
                      y1="45.72"
                      y2="8.91"
                    />
                  </g>
                </svg>
              </a>
            </div>
            <div
              class="css-1rstcrp"
            >
              <a
                class="css-1gyz2k7"
                href="https://konstantinos-lampridis-cv-documents.s3.eu-central-1.amazonaws.com/main_cv%2Bprojects.pdf"
                rel="noopener noreferrer"
                target="_blank"
              >
                'CV + Projects' 
                <svg
                  fill="none"
                  height="10px"
                  stroke="#534341"
                  stroke-width="5.108"
                  viewBox="-2.56 -2.56 69.12 69.12"
                  width="10px"
                >
                  <g
                    id="SVGRepo_iconCarrier"
                  >
                    <path
                      d="M55.4,32V53.58a1.81,1.81,0,0,1-1.82,1.82H10.42A1.81,1.81,0,0,1,8.6,53.58V10.42A1.81,1.81,0,0,1,10.42,8.6H32"
                    />
                    <polyline
                      points="40.32 8.6 55.4 8.6 55.4 24.18"
                    />
                    <line
                      x1="19.32"
                      x2="54.61"
                      y1="45.72"
                      y2="8.91"
                    />
                  </g>
                </svg>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-n1i9i7"
      >
        <div
          class="css-eqcqji"
        >
          <h3
            class="css-1td1kb3"
          >
            MSc in Artificial Intelligence
          </h3>
          <h4
            class="css-oh44f7"
          >
            University of Amsterdam
          </h4>
          <span
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            <a>
              <p
                class="css-12f1dh1"
              >
                Amsterdam, Netherlands
              </p>
              <span
                class="css-9o6sth"
              >
                2014 - 2019
              </span>
            </a>
          </span>
          <p
            class="css-1mvjx7v"
          >
            Thesis: 
            Political Spectrum Aware Topic Model
          </p>
          <div
            class="css-bvngjg"
          >
            <span
              class="css-1szrj99"
            >
              ML
            </span>
            <span
              class="css-1szrj99"
            >
              CV
            </span>
            <span
              class="css-1szrj99"
            >
              RL
            </span>
            <span
              class="css-1szrj99"
            >
              NLP
            </span>
          </div>
        </div>
        <div
          class="css-eqcqji"
        >
          <h3
            class="css-1td1kb3"
          >
            BSc in Applied Informatics
          </h3>
          <h4
            class="css-oh44f7"
          >
            University of Macedonia
          </h4>
          <span
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            <a>
              <p
                class="css-12f1dh1"
              >
                Thessaloniki, Greece
              </p>
              <span
                class="css-9o6sth"
              >
                2008 - 2013
              </span>
            </a>
          </span>
          <p
            class="css-1mvjx7v"
          >
            Thesis: 
            Computational Analysis of Simplex Points Algorithm
          </p>
          <div
            class="css-bvngjg"
          >
            <span
              class="css-1szrj99"
            >
              Operating Systems
            </span>
            <span
              class="css-1szrj99"
            >
              Linear Programming
            </span>
            <span
              class="css-1szrj99"
            >
              Discreet Mathematics
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-lpnzm1"
    >
      <div
        class="css-8her76"
        id="introduction-section"
      >
        <p
          style="font-size: 1.75rem; line-height: 1.6; margin: 1rem 0px;"
        >
          Welcome to the online space of 
          <strong>
            John Doe
          </strong>
          .
        </p>
        <p
          class="css-1ywv6q5"
        >
          Here you will find information about my
           
          <span
            style="font-weight: bold; color: rgb(65, 0, 3);"
          >
            Open Source Project Portfolio
          </span>
          ,
           
          <span
            style="font-weight: bold; color: rgb(65, 0, 3);"
          >
            Professional Experience
          </span>
          ,
           
          <span
            style="font-weight: bold; color: rgb(65, 0, 3);"
          >
            Skills
          </span>
          , and
           
          <span
            style="font-weight: bold; color: rgb(65, 0, 3);"
          >
            Education
          </span>
          .
        </p>
      </div>
      <div
        class="css-mnc2za"
        id="professional-section"
      >
        <h1
          class="css-1glvia5"
        >
          Professional Career
        </h1>
        <div
          class="css-ea9ttx"
        >
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            <h3
              class="css-1wtdklz"
            >
              Software Engineer
            </h3>
            <p
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
              Mountain View, CA
            </p>
            <p
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
              GG Navi
            </p>
            <p
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
              Sep 2022 - May 2023
            </p>
            <p
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
              I worked at GG Navi as a Software Engineer.
            </p>
            <p
              style="margin-bottom: 12px; font-family: Roboto, sans-serif; font-size: 16px;"
            >
              Activities
            </p>
            <ul
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
              <li
                style="padding: 4px 0px;"
              >
                <div
                  class="css-p4d3hp"
                >
                  I worked at GG Navi as a Software Engineer.
                </div>
              </li>
              <li
                style="padding: 4px 0px;"
              >
                <div
                  class="css-p4d3hp"
                >
                  other activity
                </div>
              </li>
            </ul>
            <div
              class="css-112f423"
            >
              <span
                class="css-e1ywo8"
              >
                python
              </span>
              <span
                class="css-e1ywo8"
              >
                docker
              </span>
            </div>
          </div>
        </div>
        <div
          class="css-yhb7v1"
        >
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            <h3
              class="css-1wtdklz"
            >
              Software Engineer
            </h3>
            <p
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
              Mountain View, CA
            </p>
            <p
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
              GG Navi
            </p>
            <p
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
              Sep 2022 - May 2023
            </p>
            <p
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
              I worked at GG Navi as a Software Engineer.
            </p>
            <p
              style="margin-bottom: 12px; font-family: Roboto, sans-serif; font-size: 16px;"
            >
              Activities
            </p>
            <ul
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
              <li
                style="padding: 4px 0px;"
              >
                <div
                  class="css-p4d3hp"
                >
                  I worked at GG Navi as a Software Engineer.
                </div>
              </li>
              <li
                style="padding: 4px 0px;"
              >
                <div
                  class="css-p4d3hp"
                >
                  other activity
                </div>
              </li>
            </ul>
            <div
              class="css-112f423"
            >
              <span
                class="css-e1ywo8"
              >
                python
              </span>
              <span
                class="css-e1ywo8"
              >
                docker
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-rc6x81"
        id="open-source-portfolio-section"
      >
        <h1
          style="background-color: rgb(255, 218, 214); color: rgb(44, 21, 19);"
        >
          Open Source & Portfolio
        </h1>
        <button>
          Reset Layout
        </button>
        <div
          class="layout react-grid-layout"
        />
      </div>
    </div>
    <div
      class="css-lsw45e"
    >
      <span
        class="css-2dse8b"
      >
        <svg
          aria-hidden="true"
          fill="#2c1513"
          height="18px"
          version="1.1"
          viewBox="0 0 16 16"
          width="18px"
        >
          <path
            d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
